### PR TITLE
test: Use mocktime to avoid intermittent failure in rpc_tests

### PR DESCRIPTION
**Pull request by maflcko**
See https://github.com/bitcoin/bitcoin/pull/21602#discussion_r611176103

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK

**Review by jonatack — COMMENTED** [review]
tACK fabc39f3563b027f5750cad2f41ecbde80dd7f5f with light code review, some comments

**Comment by jonatack** [comment]
Upgraded to tested ack.

**Comment by jonatack** [comment]
What about these
```
src/test/miner_tests.cpp:512:    SetMockTime(0);
src/test/util/setup_common.cpp:307:    SetMockTime(0);
src/test/util_tests.cpp:1406:    SetMockTime(0);
src/wallet/test/wallet_tests.cpp:299:    SetMockTime(0);
src/wallet/test/wallet_tests.cpp:382:    SetMockTime(0);
```

File: src/test/util/setup_common.cpp [hunk 1]
```diff
@@ -120,6 +120,7 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::ve
 
 BasicTestingSetup::~BasicTestingSetup()
 {
+    SetMockTime(0s);
```
Comment (jonatack) [inline src/test/util/setup_common.cpp:123]:
```suggestion
    SetMockTime(0s); // Reset mock time for other tests.
```

Comment (MarcoFalke) [reply src/test/util/setup_common.cpp]: Thanks, fixed

**Review by jarolrod — COMMENTED** [review]
ACK fa40d6a1c47ac7f3dc6c11a2e6642cfef95422c1

This gives us tighter control of time and will prevent any time slippage making this test fail. I couldn't make the banned time test fail in practice, but this prevents that.

**Comment by jonatack** [comment]
Code review ACK fa40d6a1c47ac7f3dc6c11a2e6642cfef95422c1

**Comment by MarcoFalke** [comment]
As predicted, this happens in practice:

```
test/rpc_tests.cpp(292): error: in "rpc_tests/rpc_ban": check time_remaining == banned_until - now has failed [200 != 199]
```

https://cirrus-ci.com/task/5877648200040448?logs=ci#L7931

**Pull request merged**